cloud/pkg/edgecontroller/controller/upstream.go

```diff
@@ -1008,25 +820,107 @@ func NewUpstreamController(config *v1alpha1.EdgeController, factory k8sinformer.
 		config:       *config,
 	}
 	uc.nodeLister = factory.Core().V1().Nodes().Lister()
-	uc.endpointLister = factory.Core().V1().Endpoints().Lister()
-	uc.serviceLister = factory.Core().V1().Services().Lister()
 	uc.podLister = factory.Core().V1().Pods().Lister()
 	uc.configMapLister = factory.Core().V1().ConfigMaps().Lister()
 	uc.secretLister = factory.Core().V1().Secrets().Lister()
 
-	uc.nodeStatusChan = make(chan model.Message, config.Buffer.UpdateNodeStatus)
-	uc.podStatusChan = make(chan model.Message, config.Buffer.UpdatePodStatus)
-	uc.configMapChan = make(chan model.Message, config.Buffer.QueryConfigMap)
-	uc.secretChan = make(chan model.Message, config.Buffer.QuerySecret)
-	uc.serviceAccountTokenChan = make(chan model.Message, config.Buffer.ServiceAccountToken)
-	uc.serviceChan = make(chan model.Message, config.Buffer.QueryService)
-	uc.endpointsChan = make(chan model.Message, config.Buffer.QueryEndpoints)
-	uc.persistentVolumeChan = make(chan model.Message, config.Buffer.QueryPersistentVolume)
-	uc.persistentVolumeClaimChan = make(chan model.Message, config.Buffer.QueryPersistentVolumeClaim)
-	uc.volumeAttachmentChan = make(chan model.Message, config.Buffer.QueryVolumeAttachment)
-	uc.queryNodeChan = make(chan model.Message, config.Buffer.QueryNode)
-	uc.updateNodeChan = make(chan model.Message, config.Buffer.UpdateNode)
-	uc.podDeleteChan = make(chan model.Message, config.Buffer.DeletePod)
-	uc.ruleStatusChan = make(chan model.Message, config.Buffer.UpdateNodeStatus)
+	uc.workerFactory = initWorkerFactory(uc)
+
 	return uc, nil
 }
+
+func initWorkerFactory(uc *UpstreamController) *worker.Factory {
+	workerConfigs := map[string]worker.WorkConfig{
+		model.ResourceTypeNodeStatus: {
+			Action:       "updateNodeStatus",
+			Workers:      uc.config.Load.UpdateNodeStatusWorkers,
+			Buffer:       uc.config.Buffer.UpdateNodeStatus,
+			ResourceType: model.ResourceTypeNodeStatus,
+			Handler:      uc.updateNodeStatus,
+		},
+		model.ResourceTypePodStatus: {
+			Action:       "updatePodStatus",
+			Workers:      uc.config.Load.UpdatePodStatusWorkers,
+			Buffer:       uc.config.Buffer.UpdatePodStatus,
+			ResourceType: model.ResourceTypePodStatus,
+			Handler:      uc.updatePodStatus,
+		},
+		model.ResourceTypeConfigmap: {
+			Action:       "QueryConfigMap",
```

Comment:
The format whether the first letter of `Action` is case-sendsitive should be unified.